### PR TITLE
Support special chars in url

### PR DESCRIPTION
**Pull request by abso-crichard**
We ran into an issue where we had to redirect a url that had a special character (ex. /en-us/schöttli.aspx to /en/home).
The plugin was attempting to compare schöttli with sch%C3%B6ttli so was still giving a 404 instead of successfully redirecting.
Added a Url.Encode before compare to mitigate this.

### CODE REVIEW

**Comment by abso-crichard** [comment]
I'm not sure if I merged your master branch correctly into my fork.  Is that why the checks are failing?  Thanks

**Comment by marisks** [comment]
Could you please write a test that proves that it works as expected?

File: src/Geta.404Handler/Core/CustomRedirects/CustomRedirectCollection.cs [hunk 1]
```diff
@@ -109,7 +110,7 @@ private CustomRedirect FindInternal(string url)
             // url to the <new> url.
             foreach (var redirectPair in redirectsZA)
             {
-                var oldUrl = redirectPair.Key;
+                var oldUrl = Url.Encode(redirectPair.Key);
```
Comment (marisks) [inline src/Geta.404Handler/Core/CustomRedirects/CustomRedirectCollection.cs:113]: I see the null check below that checks if `oldUrl` is not null. Does the `Url.Encode` accept null as a parameter?

**Comment by abso-crichard** [comment]
I've added a unit test for my change and changed the code slightly to fix the unit test that had been broken.

**Comment by abso-crichard** [comment]
I've updated my previous code with a new commit.  I believe all the checks are ok now.  Can you please let me know if any additional changes are required?  Thank you!

**Comment by marisks** [comment]
Test works, but I am not able to produce it in demo site. I am unable to add a letter with umlaut in the URL:
image

How did it work for you?

**Comment by absolunet-puller** [comment]
> Test works, but I am not able to produce it in demo site. I am unable to add a letter with umlaut in the URL:
> image
> 
> How did it work for you?

In our case, we are redirecting urls from a previous CMS system that did support umlauts in URLs.  We are redirecting these old urls to new urls that do not have umlauts.  I should reflect this is the test.

**Comment by marisks** [comment]
Ok. Then the new URL will always be without an umlaut. The test is fine. It will show that 404 handler supports umlauts even if Episerver does not.

**Review by marisks — APPROVED** [review]

**Pull request merged**